elixir/apps/web/test/web/controllers/sign_in_controller_test.exs

```diff
@@ -0,0 +1,37 @@
+defmodule Web.SignInControllerTest do
+  use Web.ConnCase, async: true
+
+  setup do
+    Domain.Config.put_env_override(:outbound_email_adapter_configured?, true)
+    account = Fixtures.Accounts.create_account()
+
+    {:ok, account: account}
+  end
+
+  describe "client_redirect/2" do
+    test "renders 302 with deep link location on proper cookie values", %{
+      conn: conn,
+      account: account
+    } do
+      provider = Fixtures.Auth.create_email_provider(account: account)
+      actor = Fixtures.Actors.create_actor(type: :account_admin_user, account: account)
+      identity = Fixtures.Auth.create_identity(account: account, provider: provider, actor: actor)
+
+      conn_with_cookies =
+        conn
+        |> put_client_auth_state(account, provider, identity)
+        |> get(~p"/#{account}/sign_in/client_redirect")
+
+      assert redirected = redirected_to(conn_with_cookies, 302)
+      assert redirected =~ "firezone-fd0020211111://handle_client_sign_in_callback"
```

Comment:
Can we also test here the query params? We can reuse code deleted in this PR:
```
      assert query_params = URI.decode_query(redirected_to.query)
      assert not is_nil(query_params["fragment"])
      refute query_params["fragment"] =~ "NONCE"
      assert query_params["state"] == "STATE"
      refute query_params["nonce"]
      assert query_params["actor_name"] == actor.name
      assert query_params["identity_provider_identifier"] == identity.provider_identifier
      assert client_auth_data[:state] == "STATE"
      assert not is_nil(client_auth_data[:fragment])
      assert client_auth_data[:actor_name] == Repo.preload(identity, :actor).actor.name
      assert client_auth_data[:identity_provider_identifier] == identity.provider_identifier
```

Reply:
Ah, yeah, good call.  I think when I wrote this initially it was going to a dedicated success page.  I'll get this updated.

Reply:
Updated